pkg/remote/azurerm/repository/armresources.go

```diff
@@ -0,0 +1,68 @@
+package repository
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/azcore/arm"
+	"github.com/Azure/azure-sdk-for-go/sdk/resources/armresources"
+	"github.com/cloudskiff/driftctl/pkg/remote/azurerm/common"
+	"github.com/cloudskiff/driftctl/pkg/remote/cache"
+)
+
+type ArmResourcesRespository interface {
+	ListAllResourceGroups() ([]*armresources.ResourceGroup, error)
+}
+
+type armResourcesListPager interface {
```

Comment:
You can compose the pager interface I added in virtual network PR, we'll need to merge mine first and then rebase this one.

```go

// pager.go
type pager interface {
	Err() error
	NextPage(ctx context.Context) bool
}

// armresource.go
type armResourcesListPager interface {
    pager
	PageResponse() armresources.ResourceGroupsListResponse
}
```

Reply:
You can do the rebase @sundowndev